qiskit/providers/backend_compat.py

```diff
@@ -13,185 +13,275 @@
-            except BackendPropertyError:
-                duration = None
+                    in_param["error"] = param.value
+                elif param.name == "gate_length":
+                    in_param["duration"] = apply_prefix(value=param.value, unit=param.unit)
+
+            inst_prop = InstructionProperties(**in_param)
+
+            if filter_faulty and (
+                (not properties.is_gate_operational(name, gate.qubits))
+                or any(not properties.is_qubit_operational(qubit) for qubit in gate.qubits)
+            ):
+                faulty_ops.add((name, qubits))
+                try:
+                    del prop_name_map[name][qubits]
+                except KeyError:
+                    pass
+                continue
+
+            if prop_name_map[name] is None:
+                prop_name_map[name] = {}
+            prop_name_map[name][qubits] = inst_prop
+
+        # Measure instruction property is stored in qubit property
+        measure_props = []
+
+        for qubit_idx in range(configuration.num_qubits):
+            qubit_prop = properties.qubit_property(qubit_idx)
+            in_prop = {}
+            if "readout_length" in qubit_prop:
+                in_prop["duration"] = qubit_prop["readout_length"][0]
+            elif "readout_error" in qubit_prop:
+                in_prop["error"] = qubit_prop["readout_error"][0]
+            measure_props.append(InstructionProperties(**in_prop))
+
+        prop_name_map["measure"] = {}
+        for qubit, measure_prop in enumerate(measure_props):
+            if qubit in faulty_qubits:
+                continue
+            qubits = (qubit,)
+            prop_name_map["measure"][qubits] = measure_prop
+
+    if add_delay and "delay" not in prop_name_map:
+        prop_name_map["delay"] = {
+            (q,): None for q in range(configuration.num_qubits) if q not in faulty_qubits
+        }
+
+    # Define pulse qobj converter and command sequence for lazy conversion
+    if defaults:
+        pulse_lib = list(defaults.pulse_library)
+        converter = QobjToInstructionConverter(pulse_lib)
+        for cmd in defaults.cmd_def:
+            name = cmd.name
+            qubits = tuple(cmd.qubits)
+            if (
+                name not in all_instructions
+                or name not in prop_name_map
+                or qubits not in prop_name_map[name]
+            ):
+                logger.info(
+                    "Gate calibration for instruction %s on qubits %s is found "
+                    "in the PulseDefaults payload. However, this entry is not defined in "
+                    "the gate mapping of Target. This calibration is ignored.",
+                    name,
+                    qubits,
+                )
+                continue
+
+            if (name, qubits) in faulty_ops:
+                continue
+
+            entry = PulseQobjDef(converter=converter, name=cmd.name)
+            entry.define(cmd.sequence)
```

Comment:
Defaults object has parsed calibration data in `.instruction_schedule_map` and you don't need to rebuild it.
```python
for name in inst_map.instructions:
    for qubits in inst_map.qubits_with_instruction(name):
        entry = inst_map._get_calibration_entry(name, qubits)
```
I think above code works here.

Reply:
Applied suggestion :)